jellyfin-api/src/commonMain/kotlin/org/jellyfin/sdk/api/sockets/SocketInstance.kt

```diff
@@ -174,36 +176,39 @@ public class SocketInstance internal constructor(
 			return
 		}
 
-		connection = socketConnectionFactory.create(
-			api.httpClientOptions,
-			incomingMessages,
-			outgoingMessages,
-			coroutineContext,
-		).apply {
-			val connected = connect(UrlBuilder.buildUrl(
-				baseUrl = requireNotNull(baseUrl),
-				pathTemplate = SOCKET_URL,
-				queryParameters = mapOf(
-					QUERY_DEVICE_ID to deviceInfo.id,
-					QUERY_ACCESS_TOKEN to accessToken,
-				)
-			).replace(Regex("^http"), "ws"))
-
-			if (connected) {
-				messageForwardJob?.cancel()
-				messageForwardJob = coroutineScope.launch {
-					for (message in incomingMessages) forwardMessage(message)
-				}
-				listenerHelper.activeSubscriptions.clear()
-				updateSubscriptions()
-				_state.value = SocketInstanceState.CONNECTED
-			} else {
-				_state.value = SocketInstanceState.ERROR
-				// TODO retry: failed connecting
-			}
+		connectionScope = coroutineScope + Job()
+		connectionScope!!.launch {
+			connection = socketConnectionFactory.create(
+				api.httpClientOptions,
+				incomingMessages,
+				connectionScope!!.coroutineContext,
+			).connectAndBind(connectionScope!!)
 		}
 	}
 
+	private suspend fun SocketInstanceConnection.connectAndBind(scope: CoroutineScope) = apply {
```

Comment:
Isn't the apply redundant in an extension function?
Probably better to explicitly return the connection (or something else on error).